linux/gatt/client.go

```diff
@@ -212,10 +212,30 @@ func (p *Client) ReadCharacteristic(c *ble.Characteristic) ([]byte, error) {
 }
 
 // ReadLongCharacteristic reads a characteristic value which is longer than the MTU. [Vol 3, Part G, 4.8.3]
-func (p *Client) ReadLongCharacteristic(c *ble.Characteristic) ([]byte, error) {
+func (p *Client) ReadLongCharacteristic(c *ble.Characteristic) ([]byte,error) {
 	p.Lock()
 	defer p.Unlock()
-	return nil, nil
+
+	buffer := make([]byte,512)
```

Comment:
I think it would be better not to initalize the buffer with a length here and use append to add data to the slice. 
First we don't know the length of the data to be read and second, if we use append, we don't have to keep track of the current offset.